### ISSUE TITLE
Doesnt report when using named block that doesnt exist

### ISSUE THREAD

**Issue by emattias** [typed-ember/glint#1132]
With
```
    "@glint/ember-tsc": "1.7.4",
    "@glint/template": "1.7.7",
    "@glint/tsserver-plugin": "2.5.4",
```
ember-tsc doesnt complain when using a named block that doesnt exist in a glimmer component signature. Take the `Await` component from `warp-drive`. [It doesnt have a `loading` block](https://github.com/warp-drive-data/warp-drive/blob/main/warp-drive-packages/ember/src/-private/await.gts#L49-L53) but ember-tsc still doesnt complain when I do (in a gts file):

```hbs
<Await @promise={{this.myPromise}}>
  <:loading>
    <LoadingSpinner />
  </:loading>
  {{! other code }} 
</Await>
```
